code/modules/mob/living/say.dm

```diff
@@ -250,12 +251,19 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 /mob/living/send_speech(message, message_range = 6, obj/source = src, bubble_type = bubble_icon, list/spans, datum/language/message_language=null, message_mode)
 	var/static/list/eavesdropping_modes = list(MODE_WHISPER = TRUE, MODE_WHISPER_CRIT = TRUE)
 	var/eavesdrop_range = 0
+	
 	if(eavesdropping_modes[message_mode])
 		eavesdrop_range = EAVESDROP_EXTRA_RANGE
 	var/list/listening = get_hearers_in_view(message_range+eavesdrop_range, source)
 	var/list/the_dead = list()
 	for(var/_M in GLOB.player_list)
 		var/mob/M = _M
+		if(M.stat != DEAD)

```

Comment:
check ismoth first since it's the most likely of all the conditions to result in the lowest number of items to do further checks on (resulting in slight optimisation).

i.e
```
if( (ismoth(M) && M.stat != DEAD) && get_dist(M, src) < 12 || M.z != z)
```

Reply:
or just move this check down where there is already a check for being dead,

Reply:
wrong DIP SHIT

Reply:
bruh its literaly less than 10 lines below this